src/list.rs

```diff
@@ -44,15 +42,9 @@ pub struct EncodedList {
 }
 
 pub fn cmd(opts: &Opts) -> exitcode::ExitCode {
-    let mut sources = SourceDescription::types();
-    let mut transforms = TransformDescription::types();
-    let mut sinks = SinkDescription::types();
-
-    // Remove deprecated components from list
-    let deprecated = deprecated_components();
-    sources.retain(|name| !deprecated.contains(name));
-    transforms.retain(|name| !deprecated.contains(name));
-    sinks.retain(|name| !deprecated.contains(name));
```

Comment:
Oh, we'll undoubtedly have more deprecated components in the future. This type of code won't be the way we collect them, and since `field_filter` doesn't exist... leaving it in place when it's literally going to be a no-op didn't make sense.